cpp/dolfinx/mesh/MeshTags.h

```diff
@@ -75,6 +75,28 @@ class MeshTags
   /// Move assignment
   MeshTags& operator=(MeshTags&& tags) = default;
 
+  /// Find all MeshTags for a list of values and return the corresponding
+  /// indices per value, local-to-process
+  /// @param[in] values A list values
+  /// @return The correspodning indices
+  std::vector<std::vector<std::int32_t>> indices(std::vector<T> values) const
+  {
+    std::vector<std::vector<std::int32_t>> indices;
+    for (std::int32_t i = 0; i < values.size(); ++i)
+    {
+      std::vector<std::int32_t> indices_i;
+      int n = std::count(_values.begin(), _values.end(), values[i]);
+      indices_i.reserve(n);
+      for (std::int32_t j = 0; j < _values.size(); ++j)
+      {
+        if (_values[j] == values[i])
+          indices_i.push_back(_indices[j]);
+      }
+      indices.push_back(indices_i);
```

Comment:
It would be good to restructure this `push_back` to ensure that `indices_i` is not copied.